sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_transport.py

```diff
@@ -736,4 +735,13 @@ def _write(self, s):
         See http://tools.ietf.org/html/rfc5234
         http://tools.ietf.org/html/rfc6455#section-5.2
         """
-        self.ws.send_binary(s)
+        from websocket import WebSocketConnectionClosedException, WebSocketTimeoutException
+        try:
+            self.ws.send_binary(s)
+        except WebSocketTimeoutException as e:
+            raise ConnectionError('send timed out (%s)' % e)
```

Comment:
would we want to do a `raise ConnectionError from` here and everywhere for context? maybe unnecessary but just wondering.

Reply:
wouldnt be a bad idea to do so